unit_tests/MacroTranslTests.d

```diff
@@ -0,0 +1,475 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Jun 02, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+import Common;
+
+import dstep.translator.Context;
+import dstep.translator.MacroDefinition;
+import dstep.translator.Options;
+import dstep.translator.Output;
+
+void assertTranslatesMacroDefinition(
+    string expected,
+    string source,
+    string file = __FILE__,
+    size_t line = __LINE__)
+{
+    auto translUnit = makeTranslationUnit(source);
+
+    Options options;
+    Output output = new Output;
+
+    Context context = new Context(translUnit, options);
+
+    context.macroLinkage = "";
+
+    auto children = translUnit.cursor.children(true);
+
+    assert(children.length == 1);
+
+    translateMacroDefinition(output, context, children[0]);
+
+    assertEq(expected, output.data, false, file, line);
+}
+
+private alias assertTMD = assertTranslatesMacroDefinition;
+
+// Translate basic macro definitions.
+unittest
+{
+    assertTMD(q"D
+D", q"C
+#define FOO
+C");
+
+    assertTMD(q"D
+enum FOO = 1;
+D", q"C
+#define FOO 1
+C");
+
+    assertTMD(q"D
+enum FOO = "bar";
+D", q"C
+#define FOO "bar"
+C");
+
+    assertTMD(q"D
+int FOO()
+{
+    return 0;
+}
+D", q"C
+#define FOO() 0
+C");
+
+    assertTMD(q"D
+int FOO()
+{
+    return 0;
+}
+D", q"C
+#define FOO() 0
+C");
+
+    assertTMD(q"D
+auto FOO(T)(in T a, in T b)
+{
+    return a + b;
+}
+D", q"C
+#define FOO(a, b) a + b
+C");
+
+    assertTMD(q"D
+int FOO()
+{
+    return 0 + 1;
+}
+D", q"C
+#define FOO() 0 + 1
+C");
+
+    assertTMD(q"D
+int FOO(int a)
```

Comment:
How do you determine that `a` should be `int`. Based on `1` in the body? What if a long is passed when calling the macro, that will not compile in D. Isn't using a template closer to how the macro works?